soc/arm/renesas_smartbond/da1469x/power.c

```diff
@@ -74,6 +139,8 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 			break;
 		}
 
+		gpio_latch();
```

Comment:
GPIO latching should not take place here. Peripheral blocks powered by `PD_SYS`, that should be LCDC, DMA, CRYPTO etc, should first acquire PD_SYS (by calling `da1469x_pd_acquire_noconf`) before performing an operation/transaction. That should result in increasing the associated reference counter and so `da1469x_pd_release_nowait` (called below) should return zero. In doing so, the system should enter the WFI state instead of the sleep state. Typically, an interrupt designating the end of a transaction/operation should exit the device from the idle state. With the current implementation driver's signals are freezed. GPIO latching/un-latching shoud be handled as follow:

```
		if (!da1469x_pd_release_nowait(MCU_PD_DOMAIN_SYS)) {
			barrier_dsync_fence_full();
			__WFI();
		} else {
			gpio_latch();
			da1469x_pdc_ack_all_m33();
			if (z_smartbond_sleep()) {
				sleep_state = SLEEP_STATE_SLEPT;

				if (DT_SAME_NODE(DT_PROP(DT_NODELABEL(sys_clk), clock_src),
					DT_NODELABEL(xtal32m))) {
					da1469x_clock_sys_xtal32m_switch_safe();
				}
			}
			gpio_unlatch();
		} 
```

Reply:
@ioannis-karachalios change applied